File: app/src/main/java/com/sample/android/elm/login/view/ILoginView.java
package com.sample.android.elm.login.view;

public interface ILoginView {
    void error(String errorText);

    void showProgress();

    void hideProgress();

    void showPasswordError(String errorText);

    void showLoginError(String errorText);

    void showError();

    void hideError();

    void hideLoginError();

    void hidePasswordError();

    void hideKeyboard();

    void disableLoginBtn();

    void enableLoginBtn();

    void goToMainScreen();

}

File: app/src/main/java/com/sample/android/elm/login/view/ILoginView.kt
package com.sample.android.elm.login.view

interface ILoginView {

    fun setProgress(show : Boolean)

    fun showPasswordError(errorText: String?)

    fun showLoginError(errorText: String?)

    fun setError(error: String?)

    fun hideKeyboard()

    fun setEnableLoginBtn(enabled : Boolean)

}
